src/program/packetblaster/synth/synth.lua

```diff
@@ -0,0 +1,62 @@
+-- Use of this source code is governed by the Apache 2.0 license; see COPYING.
+
+module(..., package.seeall)
+
+local engine    = require("core.app")
+local config    = require("core.config")
+local timer     = require("core.timer")
+local intel10g  = require("apps.intel.intel10g")
+local main      = require("core.main")
+local Synth     = require("apps.test.synth").Synth
+local lib       = require("core.lib")
+
+local packetblaster = require("program.packetblaster.packetblaster")
+local usage = require("program.packetblaster.synth.README_inc")
+
+local long_opts = {
+   duration     = "D",
+   help         = "h",
+   src          = "s",
+   dst          = "d",
+   sizes        = "S"
+}
+
+function run (args)
+   local opt = {}
+   local duration
+   local c = config.new()
+   function opt.D (arg) 
+      duration = assert(tonumber(arg), "duration is not a number!")  
+   end
+   function opt.h (arg)
+      print(usage)
+      main.exit(1)
+   end
+
+   local source
+   local destination
+   local sizes
+   function opt.s (arg) source = arg end
+   function opt.d (arg) destination = arg end
+   function opt.S (arg)
+     sizes = {}
+     for size in string.gmatch(arg, "%d+") do
+       sizes[#sizes+1] = tonumber(size)
+     end
+   end
+
+   args = lib.dogetopt(args, opt, "hD:s:d:S:", long_opts)
+   config.app(c, "source", Synth, { sizes = sizes,
+     src = source, dst = destination })
+
+   local patterns = args
+   local nics = packetblaster.config_loadgen(c, patterns)
```

Comment:
@mwiget With the risk of being super annoying: the code before and after uses of `config_loadgen` is identical in both cases as far as I can tell. How do you feel about moving that code into `config_loadgen` and rename it to `run_loadgen`?

E.g. instead of

```
   local patterns = args
   local nics = packetblaster.config_loadgen(c, patterns)
   assert(nics > 0, "<PCI> matches no suitable devices.")
   engine.busywait = true
   intel10g.num_descriptors = 32*1024
   engine.configure(c)
   local t = timer.new("report", packetblaster.report, 1e9, 'repeating')
   timer.activate(t)
   if duration then engine.main({duration=duration})
   else             engine.main() end
```

we would have

```
packetblaster.run_loadgen(c, args, duration)
```